frontend/src/pages/GraphPF/TrafficAnimation/TrafficPointRendererPF.tsx

```diff
@@ -0,0 +1,178 @@
+import * as React from 'react';
+import { kialiStyle } from 'styles/StyleUtils';
+import { keyframes } from 'typestyle';
+import { Edge } from '@patternfly/react-topology';
+
+export abstract class TrafficPointRenderer {
+  abstract render(
+    element: Edge,
+    animationDelay: string,
+    isInfinite: boolean,
+    onAnimationEnd?: React.AnimationEventHandler
+  ): React.SVGProps<SVGElement>;
+}
+
+function getMoveAnimation(edge: Edge, percentVisible: number, isInfinite: boolean): string {
+  const startPoint = edge.getStartPoint();
+  const endPoint = edge.getEndPoint();
+  const moveAnimation = {};
+
+  if (edge.getBendpoints().length === 0) {
+    const moveX = endPoint.x - startPoint.x;
+    const moveY = endPoint.y - startPoint.y;
+
+    moveAnimation['0%'] = { translate: '0' };
+    moveAnimation[`${percentVisible}%`] = { translate: `${moveX}px ${moveY}px`, display: isInfinite ? '' : 'none' };
+    // this acts like a delay at the end, the animation continues but nothing is visible
+    if (percentVisible < 100) {
+      moveAnimation[`${percentVisible}.1%`] = { display: 'none' };
+      moveAnimation['100%'] = { display: 'none' };
+    }
+  } else {
+    // a kiali edge can have at most 1 bendpoint, in the middle. see extendedBaseEdge.ts
+    const bendPoint = edge.getBendpoints()[0];
+    const moveBendX = bendPoint.x - startPoint.x;
+    const moveBendY = bendPoint.y - startPoint.y;
+    const moveEndX = endPoint.x - startPoint.x;
+    const moveEndY = endPoint.y - startPoint.y;
+    const bend = Math.round(percentVisible / 2);
+
+    moveAnimation['0%'] = { translate: '0' };
+    moveAnimation[`${bend}%`] = { translate: `${moveBendX}px ${moveBendY}px` };
```

Comment:
The circle and diamond should appear in the graph when they start moving.
```suggestion
    moveAnimation['0%'] = { opacity: 1, translate: '0' };
    moveAnimation[`${bend}%`] = { opacity: 1, translate: `${moveBendX}px ${moveBendY}px` };
```